src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -446,3 +467,32 @@ LocalTableEmpty(Oid tableId)
 
 	return localTableEmpty;
 }
+
+
+/* CreateTruncateTrigger creates a truncate trigger on table identified by relationId
+ * and assigns citus_truncate_trigger() as handler. The new trigger is named as
+ * citus_truncate_trigger_on_ + schemaName.tableName. Trigger name for relation my_table
+ * from schema my_schema will be citus_truncate_trigger_on_my_schema.my_table to prevent
+ * name conflicts.
+ */
+static void
+CreateTruncateTrigger(Oid relationId, char *qualifiedRelationName)
+{
+	CreateTrigStmt *trigger = NULL;
+	StringInfo triggerName = makeStringInfo();
+	appendStringInfo(triggerName, "citus_truncate_trigger_on_%s", qualifiedRelationName);
+
+	trigger = makeNode(CreateTrigStmt);
+	trigger->trigname = triggerName->data;
+	trigger->relation = NULL;
+	trigger->funcname = SystemFuncName("citus_truncate_trigger");
+	trigger->args = NIL;
+	trigger->row = false;
+	trigger->timing = TRIGGER_TYPE_BEFORE;
+	trigger->events = TRIGGER_TYPE_TRUNCATE;
+	trigger->columns = NIL;
+	trigger->whenClause = NULL;
+	trigger->isconstraint = false;
+
+	CreateTrigger(trigger, NULL, relationId, InvalidOid, InvalidOid, InvalidOid, false);
```

Comment:
tested internal trigger

- trigger is not displayed in \d+
- trigger is visible on pg_trigger table
- an integer suffix is added to trigger name, it looks like it comes from oid pool.

This will resolve the issue with naming, we do not have to do anything to prevent collisions. 
so the trigger name will be like "truncate_trigger_58380" where "_58380" is an automatically generated suffix.